evm/utils/env.py

```diff
@@ -215,4 +229,4 @@ def get(name, required=False, default=empty, type=None):
         'list': env_list,
         list: env_list,
     }.get(type, env_string)
-    return fn(name, default=default, required=required)
+    return fn(name, default=default, required=required)  # type: ignore
```

Comment:
Again, can we have a comment stating why mypy must ignore this line?

Reply:
I took another try here. With the `mypy_extensions` it should be possible to type this out as `#type: Callable[[str, DefaultNamedArg(Union[Type[empty], T], 'default'), DefaultNamedArg(bool, 'required')], T]`

However this leaves me with:

```
evm/utils/env.py:216: error: Incompatible types in assignment (expression has type "function", variable has type "Callable[[str, DefaultNamedArg(Union[Type[empty], T], 'default'), DefaultNamedArg(bool, 'required')], T]")
```

I have no idea. I'll put a comment there as to explain the `# type: ignore` for now.

Reply:
@gsalgado I finally found something to improve that. The trick was to break down the initialization of the dict from the `get(...)` call so that I could add annotations for the `Dict`. (if we didn't have to use comment annotations this could be written without breaking it apart).

https://github.com/ethereum/py-evm/pull/459/commits/27a694dfdb8e3d1422d73dc678ab02e5c0eb20ce

It's still not perfect but at least it allows us to get rid of the mypy silencing. Anyway, no need to review/merge just yet. I'll again batch a couple of changes for the next PR.